images/fcos-bastion-image/root/usr/bin/prune_nodes

```diff
@@ -0,0 +1,217 @@
+    rm -f /tmp/ssh-pass
+  else
+    echo "<4>${name}: ${bmc_address} - node responded or no pdu for this node"
+  fi
+}
+
+function wait_for_node_status() {
+  local bmc_address="${1}"
+  local bmc_user="${2}"
+  local bmc_pass="${3}"
+  local name="${4}"
+  local pdu_uri="${5}"
+  local plan="${6}"
+  local status="${7}"
+  retry_max=32
+  while [ $retry_max -gt 0 ] && ! ipmitool -I lanplus -H "${bmc_address}" -U "${bmc_user}" -P "${bmc_pass}" \
+   power status | grep -Eq "${status}"; do
+    echo "<4>Waiting for ${name}: ${bmc_address} ${plan}..."
+    sleep 15
+    retry_max=$(( retry_max - 1 ))
+  done
+  [ $retry_max -le 0 ] && echo "<3>${bmc_address} needs investigation ${plan}"
+}
+
+function wipe_disk() {
+  local bmc_address="${1}"
+  local bmc_user="${2}"
+  local bmc_pass="${3}"
+  local name="${4}"
+  local pdu_uri="${5}"
+  echo "<4>Wiping disk of ${name}: ${bmc_address}..."
+  ipmitool -I lanplus -H "${bmc_address}" -U "${bmc_user}" -P "${bmc_pass}" \
+   chassis bootparam set bootflag force_pxe options=PEF,watchdog,reset,power
+  ipmitool -I lanplus -H "${bmc_address}" -U "${bmc_user}" -P "${bmc_pass}" power reset
+}
+
+function exec_on_nodes() {
+  action=${1}
+  shift
+  params=("${@}")
+  for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+    # shellcheck disable=SC1090
+    . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+    if [ ${#bmc_address} -eq 0 ] || [ ${#bmc_user} -eq 0 ] || [ ${#bmc_pass} -eq 0 ] || [ ${#name} -eq 0 ]; then
+      echo "Error while unmarshalling hosts entries"
+      exit 1
+    fi
+
+    # Execute the action
+    "${action}" "${bmc_address}" "${bmc_user}" "${bmc_pass}" "${name}" "${pdu_uri:-}" "${params[@]}"
+
+  done
+}
+
+############
+ ## Main ##
+############
+
+if [ "$#" -ne 1 ]; then
+    echo "<3>Usage: $0 <cluster-name>"
+    return 1
+fi
+
+CLUSTER=${1}
+INTERNAL_NET=${INTERNAL_NET:-192.168.90.0/24}
+DOCKER_NET=${DOCKER_NET:-172.17.0.0/16}
+SHARED_DIR="/var/builds/${CLUSTER}"
+conf_files=(/opt/dhcpd/root/etc/dnsmasq.conf
+            /opt/bind9_zones/zone
+            /opt/bind9_zones/internal_zone.rev
+            /opt/bind9_zones/external_zone.rev
+            /etc/vips_reserved
+            /etc/hosts_pool_reserved)
+
+if [ -s "${SHARED_DIR}"/hosts.yaml ]; then
+  # Remove firewall rules for bootstrap
+  ## Remove port forwarding from bastion host to bootstrap host
+  ## Remove bootstrap host access limit to internet if in disconnected network
+  ## Remove cluster nodes access limit to internet if in disconnected network
+  firewall_cleanup "${CLUSTER}"
+
+  # Delete VLAN on Juniper
+  vlan_cleanup "${CLUSTER}"
+
+  # Reboot any stuck node pdu
+  exec_on_nodes reboot_stuck_pdu &
+
+  # Wait if a node pdu was rebooted
+  if [ "${pdu_reboot}" = "true" ]; then
+    echo "<4>Waiting for 2 min before checking node response..." && sleep 120
+    exec_on_nodes wait_for_node_status 'for pdu reboot' "' on| off'"
+  fi
```

Comment:
this check (wait_for_node_status..._) go in the reboot_stuck_pdu function, on a per-node basis.